src/selections/parser.cpp

```diff
@@ -13,93 +14,145 @@
-    {"rad2deg", [](MathAst ast){ return MathAst(new Function([](double rad){ return rad * 180 / PI; }, "rad2deg", std::move(ast)));}},
-    {"deg2rad", [](MathAst ast){ return MathAst(new Function([](double deg){ return deg * PI / 180; }, "deg2rad", std::move(ast)));}},
+    {"sin", [](MathAst ast){ return chemfiles::make_unique<Function>(static_cast<double (*)(double)>(sin), "sin", std::move(ast));}},
+    {"cos", [](MathAst ast){ return chemfiles::make_unique<Function>(static_cast<double (*)(double)>(cos), "cos", std::move(ast));}},
+    {"tan", [](MathAst ast){ return chemfiles::make_unique<Function>(static_cast<double (*)(double)>(tan), "tan", std::move(ast));}},
+    {"asin", [](MathAst ast){ return chemfiles::make_unique<Function>(static_cast<double (*)(double)>(asin), "asin", std::move(ast));}},
+    {"acos", [](MathAst ast){ return chemfiles::make_unique<Function>(static_cast<double (*)(double)>(acos), "acos", std::move(ast));}},
+    {"sqrt", [](MathAst ast){ return chemfiles::make_unique<Function>(static_cast<double (*)(double)>(sqrt), "sqrt", std::move(ast));}},
+    {"exp", [](MathAst ast){ return chemfiles::make_unique<Function>(static_cast<double (*)(double)>(exp), "exp", std::move(ast));}},
+    {"log", [](MathAst ast){ return chemfiles::make_unique<Function>(static_cast<double (*)(double)>(sqrt), "log", std::move(ast));}},
+    {"log2", [](MathAst ast){ return chemfiles::make_unique<Function>(static_cast<double (*)(double)>(log2), "log2", std::move(ast));}},
+    {"log10", [](MathAst ast){ return chemfiles::make_unique<Function>(static_cast<double (*)(double)>(log10), "log10", std::move(ast));}},
+    {"rad2deg", [](MathAst ast){ return chemfiles::make_unique<Function>([](double rad){ return rad * 180 / PI; }, "rad2deg", std::move(ast));}},
+    {"deg2rad", [](MathAst ast){ return chemfiles::make_unique<Function>([](double deg){ return deg * PI / 180; }, "deg2rad", std::move(ast));}},
 };
 
-using numeric_variable_functions_creator_t = std::function<MathAst(std::vector<Variable>)>;
+using numeric_variable_functions_creator_t = std::function<MathAst(SelectionArguments)>;
 struct NumericVariableFunction {  // NOLINT: constructor does not initialize these fields: arity
     unsigned arity;
     numeric_variable_functions_creator_t creator;
 };
 
 static std::map<std::string, NumericVariableFunction> NUMERIC_VAR_FUNCTIONS = {
-    {"distance", {2, [](std::vector<Variable> args){ return MathAst(new Distance(args[0], args[1])); }}},
-    {"angle", {3, [](std::vector<Variable> args){ return MathAst(new Angle(args[0], args[1], args[2])); }}},
-    {"dihedral", {4, [](std::vector<Variable> args){ return MathAst(new Dihedral(args[0], args[1], args[2], args[3])); }}},
-    {"out_of_plane", {4, [](std::vector<Variable> args){ return MathAst(new OutOfPlane(args[0], args[1], args[2], args[3])); }}},
+    {"distance", {2, [](SelectionArguments args) {
+        assert(args.count == 2);
+        return make_unique<Distance>(
```

Comment:
Should one use `chemfiles::make_unique` as before?
This applies for the whole map.

Reply:
Do you mean explicitly using the `chemfiles::` namespace? That should not be required, but MSVC 15 fails to perform correct ADL. It do not hurt to use it though

Reply:
>Do you mean explicitly using the chemfiles:: namespace?

Yes, I was wondering because in your other commit "Resolve ambiguous call to make_unique for MSVC" the namespace was added in a bunch of places.